tests/phpunit/tests/editor/disableContentEditorForNavigationPostType.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * @group editor
+ *
+ * @covers ::_disable_content_editor_for_navigation_post_type
+ */
+class Tests_Editor_DisableContentEditorForNavigationPostType extends WP_UnitTestCase {
+
+	const NAVIGATION_POST_TYPE     = 'wp_navigation';
+	const NON_NAVIGATION_POST_TYPE = 'wp_non_navigation';
+
+	public function tear_down() {
+		$this->enable_editor_support();
+		parent::tear_down();
+	}
+
+	/**
+	 * @ticket 56266
+	 *
+	 * @dataProvider data_should_only_disable_content_editor_for_navigation_post_types
+	 *
+	 * @param string $post_type             The post type.
+	 * @param bool   $enable_editor_support Whether to enable editor support.
+	 * @param bool   $expected              The expected result.
+	 * @param string $message               The message to display if the test fails.
+	 */
+	public function test_should_only_disable_content_editor_for_navigation_post_types( $post_type, $enable_editor_support, $expected, $message ) {
+		$post = $this->create_post( $post_type );
+		if ( $enable_editor_support ) {
+			$this->enable_editor_support();
+			$this->assertTrue( $this->supports_block_editor(), 'Editor support must be enabled before running the test.' );
+		} else {
+			$this->disable_editor_support();
+			$this->assertFalse( $this->supports_block_editor(), 'Editor support must be disabled before running the test.' );
+		}
```

Comment:
The code here does not need to be retested for each dataset. Rather, it's testing the tests set up of enabling and disabling. That can be done in a separate test methods. In doing, it reduces the code in this test and eliminates the need for a message arg to be passed.

Update coming.